src/main/java/practice/CandidateValidator.java

```diff
@@ -1,5 +1,22 @@
 package practice;
 
-public class CandidateValidator {
-    //write your code here
+import java.util.function.Predicate;
+import model.Candidate;
+
+public class CandidateValidator implements Predicate<Candidate> {
+    private static final int FIRST_YEAR_INDEX = 0;
+    private static final int LAST_YEAR_INDEX = 1;
+    private static final int MIN_AGE_CANDIDATE = 35;
+    private static final int MIN_YEAR_IN_UKRAINE = 10;
+
+    @Override
+    public boolean test(Candidate candidate) {
+        String[] firstAndLastYear = candidate.getPeriodsInUkr().split("-");
+        int periodsInUkr = Integer.parseInt(firstAndLastYear[LAST_YEAR_INDEX])
+                - Integer.parseInt(firstAndLastYear[FIRST_YEAR_INDEX]);
+        return candidate.getAge() >= MIN_AGE_CANDIDATE
+                & candidate.getNationality().equals("Ukrainian")
+                & candidate.isAllowedToVote()
+                & periodsInUkr >= MIN_YEAR_IN_UKRAINE;
```

Comment:
add a new method to find the number of years. 
for exempl 
private int periodsInUkr(Candidate candidate) { 
some logic; 
return periodInUkr;
}
```suggestion
                & periodsInUkr(candidate) >= MIN_YEAR_IN_UKRAINE;
```

Reply:
I don't think about it, thanks.